src/lib/containers/UserCardMedium.tsx

```diff
@@ -39,306 +33,306 @@ export type UserCardMediumProps = {
   isValidated?: boolean
 }
 
-export default class UserCardMedium extends React.Component<
-  UserCardMediumProps,
-  UserCardState
-> {
-  public htmlDivRef = React.createRef<HTMLDivElement>()
+/**
+ * Function handles copying to clipboard the user's email address
+ *
+ * @param {string} value the email address of the user
+ * @returns
+ */
+const copyToClipboard = (
+  htmlDivRef: React.MutableRefObject<HTMLElement | null>,
+  value: string,
+  showModal: (show: boolean) => void,
+) => (event: React.SyntheticEvent) => {
+  event.preventDefault()
+  // https://hackernoon.com/copying-text-to-clipboard-with-javascript-df4d4988697f
+  // this copies the email to the clipoard
+  const el = document.createElement('textarea')
+  el.value = value
+  el.setAttribute('readonly', '')
+  el.style.position = 'absolute'
+  el.style.left = '-9999px'
+  htmlDivRef.current!.appendChild(el)
+  el.select()
+  document.execCommand('copy')
+  htmlDivRef.current!.removeChild(el)
+  // show modal and hide after 4 seconds, the timing is per Material Design
+  showModal(true)
+  // hide after 4 seconds
+  setTimeout(() => {
+    showModal(false)
+  }, 4000)
+}
 
-  constructor(props: UserCardMediumProps) {
-    super(props)
-    this.state = {
-      showModal: false,
-      isContextMenuOpen: false,
-      ORCIDHref: undefined
-    }
-  }
+export const UserCardMedium: React.FC<UserCardMediumProps> = ({
```

Comment:
Refactored to a function component. I was having trouble reasoning with the class implementation. I ultimately didn't need to do this, but I didn't realize that until I had already done it 😬

Can revert if desired